plugins/rikki/heroeslounge/components/SlothAccount.php

```diff
@@ -461,6 +460,39 @@ public function onUpdateLinks()
         }
     }
 
+    public function onSyncDiscord()
+    {
+        if (!$user = $this->user()) {
+            return;
+        }
+
+        $sloth = SlothModel::getFromUser($user);
+
+        if (!empty($sloth->discord_id)) {
+            $newTag = Discord\Attendance::getDiscordTag($sloth->discord_id);
+
+            if (!empty($newTag)) {
+                $sloth->discord_tag = $newTag;
+                $sloth->save();
+    
+                Flash::success('Discord tag was updated successfully!');
+                if ($redirect = $this->makeRedirection()) {
+                    return $redirect;
+                }
+            } else {
+                Flash::error(`Could not update Discord tag!`);
```

Comment:
This is not a very helpful message for the user. You can add something like 'Please try again later'